src/components/trade/PriceImpact/utils.ts

```diff
@@ -64,35 +71,38 @@ function determinePriceWarning(params: PriceImpactArgs, impact: BigNumber | null
     return null
 
   // Is user's limit price less than the suggested fill price?
-  const isLimitPriceLessThanFillPrice = limitPriceBigNumber.lt(fillPrice)
+  const orderWontBeFullyExecuted = limitPriceBigNumber.lt(fillPrice)
   // Is user's limit price less than suggested fill price AND best ask?
   // One could assume that if the limit price is lower than best ask then it's lower than fill
   // but just to be safe
-  const isLimitPriceLessThanFillAndBestAsk = isLimitPriceLessThanFillPrice && limitPriceBigNumber.lt(bestAskPrice)
-
-  let warning: string
-  if (isLimitPriceLessThanFillPrice) {
-    // CASE 5: Limit price LESS THAN Fill price AND Best Ask price
-    if (isLimitPriceLessThanFillAndBestAsk) {
-      warning = '⚠️ Your order might only be filled when the market price reaches your limit price'
-    } else {
-      // CASE 3 & 4: Limit price LESS THAN Fill price but EQUAL TO or GREATER than Best Ask price
-      warning = 'Your order might only be partially filled'
-    }
-  } else {
-    // CASE 6: Limit price is GREATER THAN Fill price (and Best Ask)
-    const impactLevel = determineImpactLevel(impact)
-    // Calculate Deep Market warning threshold using slippage
-    const deepMarketPriceCeiling = limitPriceBigNumber.gte(fillPrice.times(ONE_BIG_NUMBER.plus(PLACEHOLDER_SLIPPAGE)))
-    // Check against impact level AND the slippage
-    if (impactLevel === 3 && deepMarketPriceCeiling) {
-      warning = '⚠️ Deep in the market: Your order might be fully executed at this price.'
-    } else {
-      warning = 'Your order might be fully executed at this price.'
-    }
+  const orderWontBeExecuted = limitPriceBigNumber.lt(bestAskPrice)
+  const impactLevel = determineImpactLevel(impact)
+  // Calculate Deep Market warning threshold using slippage
+  // limitPrice * 1.05
+  const isPriceAboveDeepMarketThreshold =
+    impactLevel === ImpactLevel.HIGH && limitPriceBigNumber.gte(fillPrice.times(HIGH_PLACEHOLDER_SLIPPAGE))
+
+  const isPriceMidImpactAndExecutable =
+    impactLevel >= ImpactLevel.LOW &&
+    !orderWontBeFullyExecuted &&
+    limitPriceBigNumber.gte(fillPrice.times(MID_PLACEHOLDER_SLIPPAGE))
+
+  switch (true) {
+    // CASE 6: Limit price is GREATER THAN Fill price AND upper threshold 5%
+    case isPriceAboveDeepMarketThreshold:
+      return '⚠️ Deep in the market: Your order might be fully executed at this price.'
```

Comment:
do these `⚠️`  get displayed in one of our included fonts or in a system font?
If it's a system font, not every user would necessarily have them
Probably ok, though

Reply:
ah i thought it would always be displayed but just not necessarily in our font, will it be broken on some systems?

Reply:
Font needs to support it. Or will display an empty block
But it may already be alright, if this text is displayed with the fonts we use
Just need to check

Reply:
I even have it displayed differently in Chrome and Firefox, ugh
Check diff browsers

Reply:
Firefox:
![Screen Shot 2020-10-27 at 11 46 59](https://user-images.githubusercontent.com/5121491/97277980-36256900-184a-11eb-96c2-a90b436fc3ed.png)

Chrome:
![pr1544--dexreact review gnosisdev com_trade_USDC-DAI__sell=12 price=0 93 from= expires=](https://user-images.githubusercontent.com/5121491/97277991-39b8f000-184a-11eb-98c6-54a561039993.png)

Reply:
The reason for the difference is very, very interesting:

If you copy that warning symbol and inspect it you'll have:
```js
'⚠️'.length
2
'⚠️'.charAt(1)
"️"
'⚠️'.charAt(0)
"⚠"
'⚠️'.charCodeAt(0)
9888
'⚠️'.charCodeAt(1)
65039
```
First char is the [warning symbol](https://unicode-table.com/en/26A0/). Second -- [ Variation Selector-16](https://unicode-table.com/en/FE0F/) that says _display the symbol before me as emoji_

But Chrome ignores it because of [issue](https://bugs.chromium.org/p/chromium/issues/detail?id=1033453)
That's why Firefox uses an Emoji font but Chrome does, Chrome doesn't recognize that there's an emoji

As you can't force emoji in Chrome, to have the same cross-browser experience you'll need to enforce text rendering of non emoji: `⚠` (believe you me this particular one is `length = 1`) or better yet more explicitly `&#x26A0;&#xFE0E;` (html entities), where `&#xFE0E;` is [Variation Selector-15](&#xFE0E;) to enforce non-emoji display.


Further reading [Unicode Variation Selector](https://blog.tomayac.com/2019/12/12/same-same-but-different-unicode-variation-selector-16/) , [How to prevent Unicode characters from rendering as emoji in HTML from JavaScript?](https://code-examples.net/en/q/1f6401d)

Reply:
Or just use font-awesome icons